### DISCUSSION TITLE
Typed GetContractReturnType of ABI

### DISCUSSION THREAD

**Discussion by strumswell** [wevm/viem#1714] [Question] [answered]
I was just wondering how to get a correctly typed version of a Contract via GetContractReturnType. [This](https://github.com/wevm/viem/discussions/544) does not work anymore. ("viem": "2.2.0")

**Comment by tmm** [comment, marked as answer]
There's a new syntax: [TypeScript Playground](https://www.typescriptlang.org/play?target=99#code/JYWwDg9gTgLgBAbzgUygYwEwAYCCAjYOAXzgDMoIQ4ByAQwJgE8xkB6e4AZ2oChRJYiHnDhMWcAOLIYAYQgA7GFFpoYAJWkBXKPIAqzZABphcNFGS0YyAAqa8AG2BoZj5IuMizFqwHVa9+2kXYDcYDzgACxgYMHC0TU4YSmMSckoaADcQkF5+aHgkEFpgeXlpYjIKKmos5BBWNAji+W4eHjQFRLgwO0dnV0U4AF5Tc0sbXqdg0IAKJCVaFoEYAC5I6LAZgEpiLfbO+AB3f0DZAfgRr3G-AKDzuZMVDs1FNeosAA8AOh-qOKaSmsiiUymETAslvk1vFEpQZocSgATCCHL7SCKoZCaEAAQi2KT2PERyDQ9lo5lMB0pimUqjWUlkCgWqg0MG0egMAB5wQYIKQUOhsPhgOEkD0HE41mJkHzupN+iFFABuODHW6rUS8-lq07TQZEHgAPn2NJUMC+Y0RXzw-kWaGQAHlSDMANrvD7UAC6ew6ptUX0OUGAVi+EM4pFQrvdfzgAEYMABmeTetrE0nk5DUrq+5kwDD0oJM2nqLQ6fQsbkiaWy1CYXAEcLV-nivp6mBGk25jAWixWm1k+T2p1Rz5ewlpskU33ZotmhMFxl+ktsstcnksGuC+si9cy7UnO6K9vGnPFhMBoMhsMRqAjj2GOOJ5N7IA)

**Reply by strumswell** [reply]
Thanks so much!
